core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -83,6 +87,8 @@
       ScheduledExecutorService deadlineCancellationExecutor) {
     this.method = method;
     this.callExecutor = executor;
+    // Propagate the context from the thread which initiated the call to all callbacks.
+    this.context = Context.current();
```

Comment:
I had thought we were leaning toward recording the context during `start()` instead of `newCall()`. You decided it would be better at `newCall()` to make client interception easier? I can go either way still, but I was interested in your reasoning.